### DISCUSSION TITLE
parse_error while editing XCode pbxproj in JSON Format

### DISCUSSION THREAD

**Discussion by dimitre** [nlohmann/json#3706] [Q&A] [answered]
### Description

First of all thank you for the great project!
I've recently ported the projectGenerator tool to use JSON format to edit an XCode project.

The file usually starts with a comment like this:
```
// !$*UTF8*$!
```
and it seems it cause an error in nlohmann json
```
libc++abi: terminating with uncaught exception of type nlohmann::detail::parse_error: [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - invalid literal; last read: '/'
```

and removing this line it gets another error in encoding, probably I didn't research enough to see how to parse it. 
```
libc++abi: terminating with uncaught exception of type nlohmann::detail::parse_error: [json.exception.parse_error.101] parse error at line 2, column 2: syntax error while parsing object key - invalid literal; last read: '{<U+000A><U+0009>a'; expected string literal
```

### Reproduction steps

Here is the file I'm trying to parse
it is a little unusual, even github marked as red.
https://gist.github.com/dimitre/533b60301f9186710184aff5fdc05373

### Expected vs. actual results

I know the XCode file format looks a little unusual but it would be great to be able to edit it successfully

### Error messages

```Shell
libc++abi: terminating with uncaught exception of type nlohmann::detail::parse_error: [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - invalid literal; last read: '/'


libc++abi: terminating with uncaught exception of type nlohmann::detail::parse_error: [json.exception.parse_error.101] parse error at line 2, column 2: syntax error while parsing object key - invalid literal; last read: '{<U+000A><U+0009>a'; expected string literal
```

### Compiler and operating system

macOS 12 - Xcode

### Library version

3.11.2

### Validation

- [ ] The bug also occurs if the latest version from the [`develop`](https://github.com/nlohmann/json/tree/develop) branch is used.
- [ ] I can successfully [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests).

**Comment by nlohmann** [comment, marked as answer]
You can tell the parse function to ignore comments, see https://json.nlohmann.me/features/comments/

**Comment by dimitre** [comment]
Thanks for pointing the right direction, comments solved.
Sadly @apple  is writing non standard json (in XCode project), not using the keys as literal strings.

**Reply by nlohmann** [reply]
Well, that's not looking like json at all.

**Comment by falbrechtskirchinger** [comment]
FYI, Facebook has an open-source implementation of a plist parser: https://github.com/facebookarchive/xcbuild/tree/master/Libraries/plist
This file should tell you how to use it: https://github.com/facebookarchive/xcbuild/blob/master/Libraries/plist/Tools/plutil.cpp
